plugin/evm/vm.go

```diff
@@ -745,6 +723,52 @@ func (vm *VM) writeBackMetadata() {
 	atomic.StoreUint32(&vm.writingMetadata, 0)
 }
 
+// awaitTxPoolStabilized waits for a txPoolHead channel event
+// and notifies the VM when the tx pool has stabilized to the
+// expected block hash
+// Waits for signal to shutdown from txPoolStabilizedShutdownChan chan
+func (vm *VM) awaitTxPoolStabilized() {
+	vm.shutdownWg.Add(1)
+	for {
+		select {
+		case e := <-vm.newTxPoolHeadChan.Chan():
+			switch h := e.Data.(type) {
+			case core.NewMinedBlockEvent:
+				vm.txPoolStabilizedLock.Lock()
+				if vm.txPoolStabilizedHead == h.Block.Hash() {
+					vm.txPoolStabilizedOk <- struct{}{}
+					vm.txPoolStabilizedHead = common.Hash{}
+				}
+				vm.txPoolStabilizedLock.Unlock()
+			default:
+			}
+		case <-vm.txPoolStabilizedShutdownChan:
+			vm.shutdownWg.Done()
```

Comment:
Would it make sense for this to be in a defer at the start of the function?

Reply:
Yup that would be much better